### ISSUE TITLE
Creating the frame timed out - error

### ISSUE THREAD

**Issue by theRealJeffWebber** [awslabs/amazon-quicksight-embedding-sdk#146]
Getting a "Creating the frame timed out" error after a few seconds loading a dashboard into an angular page using the embedding SDK.

**Comment by theRealJeffWebber** [comment]
This error is also present 

![image](https://user-images.githubusercontent.com/70540940/232588241-f612052d-5015-474f-902f-768edd8540fa.png)

**Comment by sonals-amazon** [comment]
Please make sure you are only embedding QuickSight's embeddable URLs i.e. the ones that contain /embed in the path. You can get an embed url using QuickSight's embedding APIs https://docs.aws.amazon.com/quicksight/latest/user/embedded-analytics-api.html or with 1-click embed code https://docs.aws.amazon.com/quicksight/latest/user/embedded-analytics-1-click.html

**Comment by theRealJeffWebber** [comment]
Thanks for the response. This is the api that I am using to get the embed URL. Is this the correct api?

response = qs.get_dashboard_embed_url(
                  AwsAccountId = accountId,
                  DashboardId = dashboardId,
                  Namespace = quicksightNamespace,
                  IdentityType = 'QUICKSIGHT',
                  UserArn = user_arn,
                  SessionLifetimeInMinutes = 600,
                  UndoRedoDisabled = undoRedoDisabled,
                  ResetDisabled = resetDisabled
            )

**Comment by theRealJeffWebber** [comment]
I did verify that the URL returned from that call does have the "embed" in it.

**Comment by bwelkin** [comment]
`Creating the frame timed out` is supposed to be thrown when the creation of the iframe is failed within 60 seconds ([code ref](https://github.com/awslabs/amazon-quicksight-embedding-sdk/blob/master/src/experiences/createExperienceFrame.ts#L178)). It may be happening around [this line](https://github.com/awslabs/amazon-quicksight-embedding-sdk/blob/b4a1f5fb4c95b7c398e34e9f13a161ad522b071f/src/commons/createIframe.ts#L130) in v2.0.3 or [here](https://github.com/awslabs/amazon-quicksight-embedding-sdk/blob/master/src/experiences/createExperienceFrame.ts#L224-L234) in v2.1.0. Could you please try placing a breakpoint and share the error message?

**Comment by theRealJeffWebber** [comment]
Thanks for the response, I set that breakpoint, but it was never hit, however, I do see something that may be contributing to the problem? There are multiple iframes in the page.

![image](https://user-images.githubusercontent.com/70540940/234921340-0426d83b-ac72-48de-85d5-d2a779349826.png)

**Comment by bwelkin** [comment]
It looks like you are calling `createEmbeddingContext` method. It is often due to state changes causing calling this multiple times in React like frameworks, or calling it for each dashboard being embedded, instead of calling it once per page.

**Comment by bwelkin** [comment]
It may not be your use case, but just as a reference info: if you are using React and `React.StrictMode`, in development mode `React.StrictMode` causes the component to mount-unmount-mount, causing `useEffect() => {...}, []);` trigger twice and potentially initializes embedding twice.

**Comment by Martina-B** [comment]
I know throwing this error is probably wanted when the dashboard takes more than 60 secs to load... It is reproducible with slower internet such as slow 3G.
But how do you catch it? Standard ErrorBoundary we use in our React app doesn't catch it and neither does wrapping the embed logic with try catch...

**Comment by pushpesh-cometchat** [comment]
> It looks like you are calling `createEmbeddingContext` method multiple times. It should only be called once. This is often due to
> 
> * state changes causing calling it multiple times in React like frameworks, or
> * calling it for each dashboard being embedded instead of calling it once per page.

@bwelkin If I am loading 3 dashboards in 3 tabs on a single page, will it still need to be called only once or once per tab? The tabs are a component unto themselves which are integrated on the page.

**Comment by kuehnmic** [comment]
@pushpesh-cometchat are you still facing this issue?

**Comment by pushpesh-cometchat** [comment]
@kuehnmic No, I modified the code as per your suggestion, to ensure the embedding context is created only once, which fixed the issue. I will close it, thanks.
